confgenerator/otel/ottl/ottl.go

```diff
@@ -0,0 +1,253 @@
+			cache.Delete(),
+			cache.Set(b),
+			statementsf(
+				`replace_pattern(%s, %q, %q)`,
+				cache,
+				`^projects/([^/]*)/traces/`,
+				"",
+			),
+		)
+		b = cache
+	}
+	statements = statements.Append(
+		statementsf(`set(%s, %s)%s`, a, b, condStr),
+	)
+	if (slices.Equal(a, LValue{"severity_text"})) {
+		// As a special case for severity_text, we need to zero out severity_number to make sure the text takes effect.
+		// TODO: Add a unit test for this.
+		statements = statements.Append(
+			LValue{"severity_number"}.SetIf(IntLiteral(0), condition),
+		)
+	}
+	return statements
+}
+
+func (a LValue) MergeMaps(source Value, strategy string) Statements {
+	return Statements{
+		statementf(`merge_maps(%s, %s, %q) where %s`, a, source, strategy, IsNotNil(source)),
+	}
+}
+
+// IsPresent returns true if the field is recursively present (with any value)
+func (a LValue) IsPresent() Value {
+	var conditions []Value
+	for i := 1; i <= len(a); i++ {
+		conditions = append(conditions, IsNotNil(a[:i]))
+	}
+	return And(conditions...)
+}
+
+func ToString(a Value) Value {
+	return valuef(`Concat([%s], "")`, a)
+}
+
+func ToInt(a Value) Value {
+	return valuef(`Int(%s)`, a)
+}
+
+func ToFloat(a Value) Value {
+	return valuef(`Double(%s)`, a)
+}
+
+func ToTime(a Value, strpformat string) Value {
+	return valuef(`Time(%s, %q)`, a, strpformat)
+}
+
+func ParseJSON(a Value) Value {
+	return valuef(`ParseJSON(%s)`, a)
+}
+
+func ConvertCase(a Value, toCase string) Value {
+	return valuef(`ConvertCase(%s, %q)`, a, toCase)
+}
+
+func IsMatch(target Value, pattern string) Value {
+	return valuef(`IsMatch(%s, %q)`, target, pattern)
+}
+
+func Equals(a, b Value) Value {
+	return valuef(`%s == %s`, a, b)
+}
+
+func Not(a Value) Value {
+	return valuef(`(not %s)`, a)
+}
+
+func And(conditions ...Value) Value {
+	var out []string
+	for _, c := range conditions {
+		out = append(out, c.String())
+	}
+	return valuef(`(%s)`, strings.Join(out, " and "))
+}
+
+func Or(conditions ...Value) Value {
+	var out []string
+	for _, c := range conditions {
+		out = append(out, c.String())
+	}
+	return valuef(`(%s)`, strings.Join(out, " or "))
+}
+
+func IsNotNil(a Value) Value {
+	return valuef(`%s != nil`, a)
+}
+
+func (a LValue) SetToBool(b Value) Statements {
+	// https://github.com/fluent/fluent-bit/blob/fd402681ad0ca0427395b07bb8a37c7c1c846cca/src/flb_parser.c#L1261
+	// "true" = true, "false" = false, else error
+	var out Statements
+	if a.String() != b.String() {
+		out = append(out, statementf(`set(%s, %s)`, a, b))
+	}
+	out = append(out,
+		statementf(`set(%s, true) where %s == "true"`, a, b),
+		statementf(`set(%s, false) where %s == "false"`, a, b),
+	)
+	return out
+}
+
+// Delete removes a (potentially nested) key from its parent maps, if that key exists.
+func (a LValue) Delete() Statements {
+	parent := a[:len(a)-1]
+	child := a[len(a)-1]
+	return Statements{
+		statementf(`delete_key(%s, %q) where %s`, parent, child, a.IsPresent()),
```

Comment:
Isn't `child` already a `Value` here (so, `%s` instead of `%q`)?

Reply:
Or I think I got it backwards -- `parent` is a string, so the `%s` should be a `%q` I think?

Reply:
Nevermind, one's a path and one's a string for some reason.